File: codeforces/go/546A.go
package main

import (
	"fmt"
)

func main() {
	var k, n, w, s int
	fmt.Scanf("%d %d %d", &k, &n, &w)
	for i := 1; i<=w; i++ {
		s += i*k
	}
	s = s-n
	if s < 0 {
		fmt.Println(0)
	} else {
		fmt.Print(s)
	}
}

File: codeforces/c/546A.c
#include<stdio.h>

int price(int k, int w) {
	int sum = 0;
	for (int i = 1; i<=w; i++) {
		sum += i*k;
	}
	return sum;
}

int main(void) {
	int k, n, w, borrowed;
	scanf("%d %d %d\n", &k, &n, &w);
	borrowed = price(k, w)-n;
	printf("%d\n", borrowed > 0 ? borrowed : 0);
	return 0;
}
